alpha/apps/kaltura/lib/myPartnerUtils.class.php

```diff
@@ -344,7 +344,7 @@ public static function getHost ( $partner_id )
 	public static function getCdnHost ( $partner_id, $protocol = null, $hostType = null )
 	{
 		// in case the request came through https, force https url
-		if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on')
+		if (infraRequestUtils::getProtocol() == 'https')
 			$protocol = 'https';
```

Comment:
I think that if you do so you can sometimes miss the protocol value given as arg for the function.
What if you will get as argument to the call $protocol=https and inside the function you will get infraRequestUtils::getProtocol() == 'http', if this case is possible you will override the given value for the protocol.

I also thought we can call to infraRequestUtils::getProtocol() only if the given protocol=null but currently we are doing the first check in getProtocol() even if the value is null and then forcing https so we will lose it in this case.  

There are multiple flows that uses this function so I think it would be less risky to use it that way and basically just add the apiProtocol check.